src/ansys/fluent/core/quantity/dimensions.py

```diff
@@ -0,0 +1,123 @@
+import ansys.fluent.core.quantity as q
+
+
+class Dimensions(object):
+    """Initialize a Dimensions object using a dimensions list or a unit string.
+
+    Parameters
+    ----------
+    units: str
+        Unit string of quantity.
+    dimensions: list
+        List of dimensions.
+
+    Returns
+    -------
+    Dimensions instance.
+    """
+
+    def __init__(
+        self, units: str = None, dimensions: list = None, unit_sys: str = None
+    ):
+        self._units_table = q.UnitsTable()
+        unit_sys = unit_sys or "SI"
+        units = units or " "
+
+        if units:
+            self._unit = units
+            self._dimensions = self._units_to_dim(units=units)
+
+        if dimensions:
+            self._dimensions, self._unit = self._dim_to_units(
+                dimensions=dimensions, unit_sys=unit_sys
+            )
+
+    def _dim_to_units(self, dimensions: list, unit_sys: str) -> str:
+        """Convert a dimensions list into a unit string.
+
+        Parameters
+        ----------
+        dimensions : list
+            List of unit dimensions.
+
+        unit_sys : str
+            Unit system of dimensions.
+
+        Returns
+        -------
+        units : str
+            Unit string representation of dimensions.
+        """
+        # Ensure dimensions list contains 9 terms
+        dimensions = [float(dim) for dim in dimensions + ((9 - len(dimensions)) * [0])]
+        sys_order = {
+            "SI": ["kg", "m", "s", "K", "radian", "mol", "cd", "A", "sr"],
+            "CGS": ["g", "cm", "s", "K", "radian", "mol", "cd", "A", "sr"],
+            "BT": ["slug", "ft", "s", "R", "radian", "slugmol", "cd", "A", "sr"],
+        }
+        units = ""
+
+        # Define unit term and associated value from dimension with dimensions list
+        for idx, dim in enumerate(dimensions):
+            if dim == 1.0:
+                units += f"{sys_order[unit_sys][idx]} "
+            elif dim != 0.0:
+                units += f"{sys_order[unit_sys][idx]}^{dim} "
+
+        return dimensions, units[:-1]
+
+    def _units_to_dim(
+        self, units: str, power: float = None, dimensions: list = None
+    ) -> list:
+        """Convert a unit string into a dimensions list.
+
+        Parameters
+        ----------
+        units : str
+            Unit string of quantity.
+        power : float
+            Power of unit string.
+
+        Returns
+        -------
+        dimensions : list
+            Dimensions representation of unit string.
+        """
+
+        power = power or 1.0
+        dimensions = dimensions or [0.0] * 9
+
+        # Split unit string into terms and parse data associated with individual terms
+        for term in units.split(" "):
+            _, unit_term, unit_term_power = self._units_table.filter_unit_term(term)
+
+            unit_term_power *= power
```

Comment:
I'm not sure about this `power` argument appearing in this method. If your output is a dimensions list then `power` can simply act as a multiplier mapped over the whole list. There really is no need to involve `power` prior to that point. It would be good to have a simple method only responsible for getting the `dimensions` for `units`.

Reply:
`power` is only applied to the terms that compose a derived unit string. 

For example: `m farad^-2`: 
- "m" is a fundamental unit and it's power in this case is 1 so nothing changes. 
- "farad^-2" becomes ("N m V^-2")^-2
- ("N V^-2")^-2 becomes ("kg m s^-2 m A ohm")^-2
- ("kg m s^-2 m A ohm")^-2 is simplified to "kg^2 m^4 s^-8 A^-4"
- the final SI unit string: `m^5 kg^2 s^-8 A^-4`

If power is mapped over the whole list then the `^-2` from farad in this case would be applied to the first `m` within the unit string even though the `^-2` is only supposed to apply to the terms derived from within `farad`.